src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTestTransformer.java

```diff
@@ -147,21 +170,32 @@ private GapicSurfaceTestClassView createTestClassView(SurfaceTransformerContext
 
   private List<GapicSurfaceTestCaseView> createTestCaseViews(SurfaceTransformerContext context) {
     ArrayList<GapicSurfaceTestCaseView> testCaseViews = new ArrayList<>();
+    TestCaseNameTable nameTable = new TestCaseNameTable();
     for (Method method : context.getNonStreamingMethods()) {
       MethodTransformerContext methodContext = context.asMethodContext(method);
-      testCaseViews.add(createTestCaseView(methodContext));
+      MethodConfig methodConfig = methodContext.getMethodConfig();
+      if (methodConfig.isFlattening()) {
+        for (List<Field> paramFields : methodConfig.getFlattening().getFlatteningGroups()) {
+          testCaseViews.add(createTestCaseView(methodContext, paramFields, nameTable));
+        }
+      } else {
+        // TODO(shinfan): Add support of non-flattening method
```

Comment:
I consider this as an addition to the existing functionality so I would prefer to not interrupt the execution even there is a non-flattening method.  Alternatively I could print a warning message here. Do you think it would be sufficient?